Tools/CISupport/ews-build/steps.py

```diff
@@ -1462,6 +1462,38 @@ class UnApplyPatch(CleanWorkingDirectory):
     name = 'unapply-patch'
     descriptionDone = ['Unapplied patch']
 
+    def doStepIf(self, step):
+        return self.getProperty('patch_id')
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+
+class RevertPullRequest(steps.ShellSequence):
+    name = 'revert-pull-request'
+    description = ['revert-pull-request running']
+    descriptionDone = ['Reverted pull request']
+    flunkOnFailure = True
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(RevertPullRequest, self).__init__(timeout=5 * 60, **kwargs)
+
+    def run(self):
+        self.commands = []
+        for command in [
+            ['git', 'clean', '-f', '-d'],
+            ['git', 'checkout', self.getProperty('github.base.sha')],
+        ]:
+            self.commands.append(util.ShellArg(command=command, logname='stdio'))
+        return super(RevertPullRequest, self).run()
+
+    def doStepIf(self, step):
+        return self.getProperty('github.base.sha')
```

Comment:
Probably should use 'github.number' property here, since that is what we use to determine if this is a Pull Request workflow.
If it is a PR workflow and it doesn't have github.base.sha property, it's better to fail here instead of skipping the step.